File: surf-api-bukkit/surf-api-bukkit-server/src/main/java/dev/slne/surf/surfapi/bukkit/server/impl/nms/bridges/SurfBukkitNmsStatsBridgeImpl.java
package dev.slne.surf.surfapi.bukkit.server.impl.nms.bridges;

import static com.google.common.base.Preconditions.checkNotNull;

import dev.slne.surf.surfapi.bukkit.api.nms.bridges.SurfBukkitNmsStatsBridge;
import dev.slne.surf.surfapi.bukkit.server.nms.NmsUtil;
import dev.slne.surf.surfapi.bukkit.server.reflection.Reflection;
import javax.annotation.ParametersAreNonnullByDefault;
import org.bukkit.entity.Player;

@ParametersAreNonnullByDefault
public final class SurfBukkitNmsStatsBridgeImpl implements SurfBukkitNmsStatsBridge, NmsUtil {

  @Override
  public String getPlayerStatsAsJson(Player player) {
    checkNotNull(player, "player");

    return Reflection.SERVER_STATS_COUNTER_PROXY.toJson(toNms(player).getStats());
  }

  @Override
  public void savePlayerStatsToFile(Player player) {
    checkNotNull(player, "player");

    toNms(player).getStats().save();
  }
}

File: surf-api-bukkit/surf-api-bukkit-server/src/main/kotlin/dev/slne/surf/surfapi/bukkit/server/impl/nms/bridges/SurfBukkitNmsStatsBridgeImpl.kt
package dev.slne.surf.surfapi.bukkit.server.impl.nms.bridges

import com.google.auto.service.AutoService
import dev.slne.surf.surfapi.bukkit.api.nms.NmsUseWithCaution
import dev.slne.surf.surfapi.bukkit.api.nms.bridges.SurfBukkitNmsStatsBridge
import dev.slne.surf.surfapi.bukkit.server.nms.toNms
import dev.slne.surf.surfapi.bukkit.server.reflection.Reflection
import dev.slne.surf.surfapi.core.api.util.checkInstantiationByServiceLoader
import org.bukkit.entity.Player

@AutoService(SurfBukkitNmsStatsBridge::class)
@NmsUseWithCaution
class SurfBukkitNmsStatsBridgeImpl : SurfBukkitNmsStatsBridge {
    init {
        checkInstantiationByServiceLoader()
    }

    override fun getPlayerStatsAsJson(player: Player): String {
        return Reflection.SERVER_STATS_COUNTER_PROXY.toJson(player.toNms().stats)
    }

    override fun savePlayerStatsToFile(player: Player) {
        player.toNms().stats.save()
    }
}
